pkg/hostpath/hostpath.go

```diff
@@ -564,27 +618,32 @@ func deleteIfEmpty(name string) {
 	}
 }
 
+func (hp *hostPath) innerDeleteHostpathVolume(top string) {
+	// reminder, path is filepath.Join(DataRoot, [anchor-dir | bind-dir], volID, podNamespace, podName, podUID, podSA)
+	// delete SA dir
+	err := os.RemoveAll(top)
+	if err != nil {
+		klog.Warningf("error deleting %s: %s", top, err.Error())
+	}
+	currentLocation := top
+	// we deleteIfEmpty on the remaining 4 levels
+	for i := 0; i < 4; i++ {
+		parentDir := filepath.Dir(currentLocation)
+		deleteIfEmpty(parentDir)
+		currentLocation = parentDir
+	}
+}
+
 // deleteVolume deletes the directory for the hostpath volume.
 func (hp *hostPath) deleteHostpathVolume(volID string) error {
 	klog.V(4).Infof("deleting hostpath volume: %s", volID)
 
-	hpv := getHPV(volID)
+	hpv := hp.getHostpathVolume(volID)
 	if hpv != nil {
```

Comment:
```suggestion
	if hpv := hp.getHostpathVolume(volID); hpv != nil {
```
I'm not 100% sure this one will work, but I think so?

Reply:
yeah seems like it should